front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -228,14 +230,17 @@ export function updateDaySinceDeparture(
           daySinceDeparture: currentDaySinceDeparture,
         };
       }
-      previousTime = arrivalInSeconds;
+      previousTime = isMidnight ? 0 : arrivalInSeconds;
     }
 
     let formattedDeparture: TimeExtraDays | undefined;
-    if (stopFor && arrivalInSeconds) {
+
+    if (stopFor && arrivalInSeconds !== null) {
       const departureInSeconds = (arrivalInSeconds + Number(stopFor)) % SECONDS_IN_A_DAY;
       const isAfterMidnight = departureInSeconds < previousTime;
```

Comment:
I was wondering if this is really the right way to figure out whether the day changed. It would be nicer to strip the time from the date+time, and then compare that with the previous one.

Reply:
This is indeed an orthogonal suggestion but also a good idea, maybe we could create an enhancement/refactor ticket ?

Reply:
Sure thing